pages/partials/navbar.ejs

```diff
@@ -9,136 +20,416 @@
+                <div class="col-1 px-0"><% if (authnViewTypeMenuChecked == 'student-no-rules') { %>&check;<% } %></div>
+                <div class="col pl-0">Student page without access rules</div>
+              </div>
+            </a>
+
+            <a class="dropdown-item" href="<%= studentLink %>" id="navbar-user-view-authn-student">
+              <div class="d-flex flex-row justify-content-left">
+                <div class="col-1 px-0"><% if (authnViewTypeMenuChecked == 'student') { %>&check;<% } %></div>
+                <div class="col pl-0">Student page with access rules</div>
+              </div>
+            </a>
+
+            <% if (authz_data.authn_user.uid != authz_data.user.uid) {
+
+              var headingViewTypeMenu = 'View type';
+              if (authz_data.user.name) {
+                  headingViewTypeMenu += ` for ${authz_data.user.name} (${authz_data.user.uid})`;
+              } else {
+                headingViewTypeMenu += ` for ${authz_data.user.uid}`;
+              }
+
+              var viewTypeMenuChecked = '';
+              if (viewType == 'instructor') {
+                if (authz_data.has_course_permission_preview || authz_data.has_course_instance_permission_view) {
+                  viewTypeMenuChecked = 'instructor'
+                }
+              } else {
+                if (authz_data.has_course_permission_preview || authz_data.has_course_instance_permission_view) {
+                  viewTypeMenuChecked = 'student-no-rules'
+                } else {
+                  viewTypeMenuChecked = 'student'
+                }
+              }
+            %>
+
+              <div class="dropdown-divider"></div>
+              <h6 class="dropdown-header"><%= headingViewTypeMenu %></h6>
+
+              <% if (authz_data.user_with_requested_uid_has_instructor_access_to_course_instance) { %>
+                <a class="dropdown-item" href="<%= instructorLink %>" id="navbar-user-view-instructor">
+                  <div class="d-flex flex-row justify-content-left">
+                    <div class="col-1 px-0"><% if (viewTypeMenuChecked == 'instructor') { %>&check;<% } %></div>
+                    <div class="col pl-0">Instructor page</div>
+                  </div>
+                </a>
               <% } %>
+
+              <a class="dropdown-item" href="<%= studentLink %>" id="navbar-user-view-student-no-rules">
+                <div class="d-flex flex-row justify-content-left">
+                  <div class="col-1 px-0"><% if (viewTypeMenuChecked == 'student-no-rules') { %>&check;<% } %></div>
+                  <div class="col pl-0">Student page without access rules</div>
+                </div>
+              </a>
+
+              <a class="dropdown-item" href="<%= studentLink %>" id="navbar-user-view-student">
+                <div class="d-flex flex-row justify-content-left">
+                  <div class="col-1 px-0"><% if (viewTypeMenuChecked == 'student') { %>&check;<% } %></div>
```

Comment:
This case (view as effective user in student mode with access rules) isn't working for me. Whenever I select it, I instead get "view as effective user in student mode _without_ access rules".

Reply:
This turned out to be because I was testing in the example course, it had a special behavior where it always granted Course Content Viewer to every user, for both the authn user and the effective user. I changed this (see 0fe1c3e557b934ad4a4bf462f6da8d7992a3426c below) so that the example course override only applies to the authn user. This lets us play with effective student users (with access restrictions) in the example course.